src/main/java/com/gregtechceu/gtceu/api/data/worldgen/generator/veins/CuboidVeinGenerator.java

```diff
@@ -0,0 +1,285 @@
+                material -> Stream.of(Either.<BlockState, Material>right(material)))
+            .forEach(entry -> result.add(Map.entry(entry, 2)));
+        middle.map(blockStates -> blockStates.stream().map(state -> Either.<BlockState, Material>left(state.state)),
+                material -> Stream.of(Either.<BlockState, Material>right(material)))
+            .forEach(entry -> result.add(Map.entry(entry, 3)));
+        bottom.map(blockStates -> blockStates.stream().map(state -> Either.<BlockState, Material>left(state.state)),
+                material -> Stream.of(Either.<BlockState, Material>right(material)))
+            .forEach(entry -> result.add(Map.entry(entry, 4)));
+        spread.map(blockStates -> blockStates.stream().map(state -> Either.<BlockState, Material>left(state.state)),
+                material -> Stream.of(Either.<BlockState, Material>right(material)))
+            .forEach(entry -> result.add(Map.entry(entry, 7)));
+        return result;
+    }
+
+    @Override
+    public Map<BlockPos, OreBlockPlacer> generate(WorldGenLevel level, RandomSource random, GTOreDefinition entry, BlockPos origin, ChunkPos currentChunk) {
+        Map<BlockPos, OreBlockPlacer> generatedBlocks = new Object2ObjectOpenHashMap<>();
+
+        // determine density based on distance from the origin chunk
+        // this makes the vein more concentrated towards the center
+        double xLength = currentChunk.x - origin.getX();
+        double zLength = currentChunk.z - origin.getZ();
+        double volume = Math.sqrt(2 + (xLength * xLength) + (zLength * zLength));
+
+        int size = entry.clusterSize();
+
+        int westBound = origin.getX() - random.nextInt(size);
+        int eastBound = origin.getX() + random.nextInt(size);
+        int northBound = origin.getZ() - random.nextInt(size);
+        int southBound = origin.getZ() + random.nextInt(size);
+
+        int minY = this.minY;
+        int startY = minY + random.nextInt(this.maxY - minY - 5);
+
+        int localDensity = (int) Math.max(1, entry.density() * 0.1 * volume);
+
+        int topAmount = 0;
+        int middleAmount = 0;
+        int bottomAmount = 0;
+        int spreadAmount = 0;
+
+        BlockPos.MutableBlockPos pos = new BlockPos.MutableBlockPos();
+        for (int layerOffset = -1; layerOffset <= 7; layerOffset++) {
+            int layer = startY + layerOffset;
+            if (level.isOutsideBuildHeight(layer))
+                continue;
+            for (int x = westBound; x < eastBound; x++) {
+                int weightX = Math.max(1, Math.max(Mth.abs(westBound - x), Mth.abs(eastBound - x)) / localDensity);
+                for (int z = northBound; z < southBound; z++) {
+                    int weightZ = Math.max(1, Math.max(Mth.abs(southBound - z), Mth.abs(northBound - z)) / localDensity);
+
+                    final var randomSeed = random.nextLong(); // Fully deterministic regardless of chunk order
+                    pos.set(x, layer, z);
```

Comment:
This is probably the spot where you want to compute the distance from the center instead, and use that for modifying the density ^^